### PR TITLE
drivers: eth: native_posix: Allow non-root access

### PR DESCRIPTION
**Pull request by jukkar**
Change the default behaviour of the host network interface
setup. Now user needs to execute net-setup.sh script from
net-tools project to setup host ethernet interface. This way
zephyr.exe can be started as a normal user.

Old behaviour is still there if one enables
    CONFIG_ETH_NATIVE_POSIX_STARTUP_AUTOMATIC=y

in which case one needs to use the command
    sudo --preserve-env zephyr.exe

to start the Zephyr process.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

### CODE REVIEW

**Comment by jukkar** [comment]
See also related PR at https://github.com/zephyrproject-rtos/net-tools/pull/22

File: drivers/ethernet/Kconfig.native_posix
Comment (pfalcon) [inline drivers/ethernet/Kconfig.native_posix]: I guess "annoying" is not the word we want to have in technical reference docs (where all this stuff goes, besides just "make menuconfig").

E.g., could say "This can be no what user wants and would require executing the zephyr.exe as a root user". Otherwise, calling to @dbkinder for suggestions.

File: drivers/ethernet/Kconfig.native_posix
Comment (pfalcon) [inline drivers/ethernet/Kconfig.native_posix]: Typo, should be "privileges".

File: drivers/ethernet/Kconfig.native_posix
Comment (rmstoi) [reply drivers/ethernet/Kconfig.native_posix]: I agree with "annoying" wording issue. I see it more as a security risk/vulnerability.

File: drivers/ethernet/Kconfig.native_posix
Comment (rmstoi) [inline drivers/ethernet/Kconfig.native_posix]: One benefit of starting the interface manually is that the interface is permanent, meaning it remains after echo server is stopped. The advantage of this is that wireshark will be left running between zephyr restarts and initial packets will be captured too. Hence, using CONFIG_ETH_NATIVE_POSIX_STARTUP_SCRIPT to start wireshark won't be needed. IMHO this benefit could be mentioned too.

Comment (pfalcon) [reply drivers/ethernet/Kconfig.native_posix]: Well, given that the manual start is new default, and this help text is already long, I guess we shouldn't add wireshark howto here.

(I for example would think that it this description "thumbs down" running as root, it should mention that net-setup.sh will certainly require root too - but well, we don't write a magazine article here.)

**Comment by jukkar** [comment]
Fixed the sanitychecker, and fixed documentation according to comments.

File: samples/net/eth_native_posix/README.rst
Comment (dbkinder) [inline samples/net/eth_native_posix/README.rst]:
```suggestion
If the :option:`CONFIG_ETH_NATIVE_POSIX_STARTUP_AUTOMATIC` option
is not enabled (this is the default), then the user should
```

File: samples/net/eth_native_posix/README.rst
Comment (dbkinder) [inline samples/net/eth_native_posix/README.rst]:
```suggestion
will create the zeth interface and set up IP addresses and routes.
```

Comment (dbkinder) [reply samples/net/eth_native_posix/README.rst]: FYI, ``setup`` is a noun or adjective (e.g., the setup script), while ``set up`` is the verb form.

File: samples/net/eth_native_posix/README.rst
Comment (dbkinder) [inline samples/net/eth_native_posix/README.rst]:
```suggestion
By running ``net-setup.sh`` manually, you can run the Zephyr process as a non-root
```

File: samples/net/eth_native_posix/README.rst
Comment (dbkinder) [inline samples/net/eth_native_posix/README.rst]:
```suggestion
user. This is the recommended and preferred way.
```

File: drivers/ethernet/Kconfig.native_posix
Comment (dbkinder) [inline drivers/ethernet/Kconfig.native_posix]: Seems like too much information here, and overlaps with eth_native_posix/README.rst.
How about:

    If set, the native_posix ethernet driver will set up the network interface,
    requiring ``zephyr.exe`` to be run with root privileges (needed to create and
    configure the TAP device).  

    If not set (the default and recommended way),
    the network interface must be set up manually using ``net-setup.sh`` (from
    the net-tools project repo). The ``zephyr.exe`` program can then be run
    as a non-root user. 

And move the other details here over to the eth_native_posix/README.rst doc.

File: samples/net/eth_native_posix/README.rst
Comment (dbkinder) [reply samples/net/eth_native_posix/README.rst]: After this, add some of the details that were in the Kconfig file:

    When the network interface is set up manually, you can leave the wireshark
    to monitor the interface, and then start and stop the zephyr process without
    stopping the wireshark. 

    Setting things manually works the same as working with SLIP
    connectivity in QEMU.

File: samples/net/eth_native_posix/README.rst
Comment (dbkinder) [reply samples/net/eth_native_posix/README.rst]: Continue with the command example that was in the Kconfig file:

    You can run the ``net-setup.sh`` script like this::

       cd net-tools
       ./net-setup.sh

    or::

       ./net-setup.sh --config ./zeth-vlan.conf

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: samples/net/eth_native_posix/README.rst
Comment (pfalcon) [reply samples/net/eth_native_posix/README.rst]: The irony is that net-setup.sh itself needs root. @jukkar's wording seems to be carefully crafted to blend that fact into background ;-). So, let me suggest my edition too after all:

~~~
While running ``net-setup.sh`` requires root access (once), afterwards 
Zephyr processes can be run as a non-root user.
~~~

**Comment by jukkar** [comment]
Comments applied to documentation.

**Comment by pfalcon** [comment]
Looks good to me, but would like to give @aescolar priority to cast his vote first.

**Comment by aescolar** [comment]
I did take a quick look yesterday and it seemed sensible to me. But I wasn't too thorough, so I preferred to not give a +2.

File: samples/net/eth_native_posix/README.rst
Comment (dbkinder) [inline samples/net/eth_native_posix/README.rst]:
```suggestion
   sudo ./net-setup.sh
```

File: samples/net/eth_native_posix/README.rst
Comment (dbkinder) [inline samples/net/eth_native_posix/README.rst]:
```suggestion
   sudo ./net-setup.sh --config ./zeth-vlan.conf
```

**Review by dbkinder — DISMISSED** [review]
add use of sudo when running net-setup.sh

**Comment by pfalcon** [comment]
@aescolar : So, if you don't see big issues with this (like, it doesn't break your usecases), I guess it only makes sense to move on with this, so let me approve from my side.

**Review by pfalcon — APPROVED** [review]

**Comment by aescolar** [comment]
@pfalcon : I dont have any use case myself for the ethernet driver. I just reviewed with the native_posix maintainer hat on.

**Comment by pfalcon** [comment]
@aescolar: Good, thanks for confirming. (Just in case, my leaning towards getting explicit ok from you is my reaction to a conversion we had lately on IRC re: potentially modifying native_port in such a way that it breaks, or makes problematic, your own usecases. I'm sorry in advance if I'm now overcautious on that. (And a kind of apology for will have involved you in another ticket I'm about to submit ;-) )).

File: samples/net/eth_native_posix/README.rst
Comment (jukkar) [reply samples/net/eth_native_posix/README.rst]: There is no need to put sudo here. The script will do sudo automatically if user does not put it there.

**Review by rmstoi — APPROVED** [review]

**Pull request merged**
